### PR TITLE
Feature/71 gpt as data source

### PR DESCRIPTION
**Pull request by soapyheas**
Adding the GPT summarizer capability, as well as a more robust call to the detailed places API, formatted in the google_places step.

Issue #105: Business type has been able to be retrieved, but is not comprehensive. Refer to issue for further info.
Pull request #109: This pull request might conflict. #109 calls the detailed Google API again.

### CODE REVIEW

**Review by rbbozkurt — APPROVED** [review]
Good job! Everything looked fine to me! Only a comment on `google_place.py`, i believe with the use of more steps in Pipeline, we need to be careful about the possibility of this file becoming too large and complex. It might be better to split it.

File: src/bdc/steps/google_places.py
Comment (luccalb) [inline src/bdc/steps/google_places.py:195]: Maybe merge felixs' branch into yours first, as he has implemented a logger module already

File: src/bdc/steps/google_places.py
Comment (luccalb) [inline src/bdc/steps/google_places.py:236]: this should probably not happen here, but rather in pipeline.py or a seperate step

File: src/bdc/steps/gpt_summarizer.py
Comment (luccalb) [inline src/bdc/steps/gpt_summarizer.py:20]: Nice work!

**Review by luccalb — CHANGES_REQUESTED** [review]
Nice work overall! Please be sure to address the merge conflicts first, also check Felixs' implementation of the logger module and wait for him to merge first, or merge his branch directly into yours. Otherwise, as the logger change affects most files, this could become a source of more merge conflicts.

File: src/bdc/steps/google_places.py
Comment (felix-zailskas) [inline src/bdc/steps/google_places.py:41]: After the merge with the new dev make sure that this class still has a correct added_cols attribute.

File: src/bdc/steps/gpt_summarizer.py
Comment (felix-zailskas) [inline src/bdc/steps/gpt_summarizer.py:20]: Again will need the added_cols attribute after merge with dev

**Review by felix-zailskas — COMMENTED** [review]
NIce work, just make sure to include the added_cols attribute in the steps and replace ols prints/logs to use the new logger after merging with dev

File: src/bdc/steps/gpt_summarizer.py
Comment (soapyheas) [reply src/bdc/steps/gpt_summarizer.py:46]: For some reason it doesn't seem to take axis as a parameter, but seems to work without. We discussed this the other day, and think that it automatically sets axis as 1. @rbbozkurt do you have any thoughts?

File: src/bdc/steps/google_places.py
Comment (soapyheas) [reply src/bdc/steps/google_places.py:220]: Yep, he merged before me, so I just handled it in the merge conflicts

File: src/bdc/steps/gpt_summarizer.py
Comment (luccalb) [reply src/bdc/steps/gpt_summarizer.py:46]: I think it works because you select only the column "google_places_website" which only has one axis, which is row by row.  
Alternative would be like this:
```suggestion
       self.df[self.extracted_col_name_website_summary] = self.df.progress_apply(lambda lead:
           self.summarize_the_company_website(lead["google_places_website"]), axis=1)
```

File: src/bdc/pipeline.py
Comment (felix-zailskas) [inline src/bdc/pipeline.py:64]: Redundant prints

File: src/bdc/pipeline.py
Comment (luccalb) [inline src/bdc/pipeline.py:63]: Pls remove these prints

File: src/bdc/steps/google_places.py
Comment (felix-zailskas) [inline src/bdc/steps/google_places.py:45]: These also need to be inside of the added_cols

File: src/bdc/steps/google_places.py
Comment (felix-zailskas) [inline src/bdc/steps/google_places.py:203]: Change to log.error

File: src/bdc/steps/google_places.py
Comment (felix-zailskas) [inline src/bdc/steps/google_places.py:207]: log.error

File: src/bdc/steps/google_places.py
Comment (luccalb) [inline src/bdc/steps/google_places.py:203]:
```suggestion
                log.warning(f"Failed to fetch data. Status code: {response.get('status')}")
```

File: src/bdc/steps/google_places.py
Comment (felix-zailskas) [inline src/bdc/steps/google_places.py:215]: log.error

File: src/bdc/steps/google_places.py
Comment (luccalb) [inline src/bdc/steps/google_places.py:207]:
```suggestion
            log.warning(f"Error: {str(e)}")
```

File: src/bdc/steps/google_places.py
Comment (luccalb) [inline src/bdc/steps/google_places.py:215]:
```suggestion
            log.warning(f"Error: {error_message}")
```

**Review by felix-zailskas — CHANGES_REQUESTED** [review]

**Review by luccalb — APPROVED** [review]
Thanks for the changes! Just found some old prints you might've missed.

File: src/bdc/steps/gpt_summarizer.py
Comment (soapyheas) [reply src/bdc/steps/gpt_summarizer.py]: Yep I see, changed to your suggestion for consistency with the other steps

**Comment by soapyheas** [comment]
Added all requested changes :) @felix-zailskas

**Pull request merged**
